Clarify behavior when decommissioningMaxPercentOfMaxSegmentsToMove = 0 (#13157)
apache/druid: docs/configuration/index.md

|`killPendingSegmentsSkipList`|List of data sources for which pendingSegments are _NOT_ cleaned up if property `druid.coordinator.kill.pendingSegments.on` is true. This can be a list of comma-separated data sources or a JSON array.|none|
|`maxSegmentsInNodeLoadingQueue`|The maximum number of segments that could be queued for loading to any given server. This parameter could be used to speed up segments loading process, especially if there are "slow" nodes in the cluster (with low loading speed) or if too much segments scheduled to be replicated to some particular node (faster loading could be preferred to better segments distribution). Desired value depends on segments loading speed, acceptable replication time and number of nodes. Value 1000 could be a start point for a rather big cluster. Default value is 100. |100|
|`decommissioningNodes`| List of historical servers to 'decommission'. Coordinator will not assign new segments to 'decommissioning' servers,  and segments will be moved away from them to be placed on non-decommissioning servers at the maximum rate specified by `decommissioningMaxPercentOfMaxSegmentsToMove`.|none|

> |`decommissioningMaxPercentOfMaxSegmentsToMove`| Upper limit of segments the Coordinator can move from decommissioning servers to active, non-decommissioning, servers during a single run. This value is relative to the total maximum segment movements allowed during one run based upon the value of `maxSegmentsToMove`.<br><br>If `decommissioningMaxPercentOfMaxSegmentsToMove` is 0, the Coordinator does not move segments to decommissioning servers, effectively putting them in a type of "maintenance" mode. In this case, decommissioning servers do not participate in balancing or assignment by load rules. The Coordinator still considers segments on decommissioning servers as candidates to move or replicate on active servers.<br><br>Decommissioning can stall if there are no available active servers to place the segments. You can use the maximum percent of decommissioning segment movements to prioritize balancing or to decrease commissioning time to prevent active servers from being overloaded. The value should be between 0 and 100.|70|

Reviewer: To clarify, when `decommissioningMaxPercentOfSegmentsToMove` is 0, segments on decommissioning servers are not considered eligible for move.

Suggestions:
1. The Coordinator still considers segments on decommissioning servers as candidates to ~~move or~~ replicate on active servers.
2. Decommissioning can stall if there is no available active server to ~~place the segments~~ move segments to. ("place" is ambiguous as it might refer to both replication or balancing)

> |`decommissioningMaxPercentOfMaxSegmentsToMove`| Upper limit of segments the Coordinator can move from decommissioning servers to active, non-decommissioning, servers during a single run. This value is relative to the total maximum segment movements allowed during one run based upon the value of `maxSegmentsToMove`.<br><br>If `decommissioningMaxPercentOfMaxSegmentsToMove` is 0, the Coordinator does not move segments to decommissioning servers, effectively putting them in a type of "maintenance" mode. In this case, decommissioning servers do not participate in balancing or assignment by load rules. The Coordinator still considers segments on decommissioning servers as candidates to move or replicate on active servers.<br><br>Decommissioning can stall if there are no available active servers to place the segments. You can use the maximum percent of decommissioning segment movements to prioritize balancing or to decrease commissioning time to prevent active servers from being overloaded. The value should be between 0 and 100.|70|

Reviewer 2: ```suggestion
|`decommissioningMaxPercentOfMaxSegmentsToMove`| Upper limit of segments the Coordinator can move from decommissioning servers to active non-decommissioning servers during a single run. This value is relative to the total maximum number of segments that can be moved at any given time based upon the value of `maxSegmentsToMove`.<br /><br />If `decommissioningMaxPercentOfMaxSegmentsToMove` is 0, the Coordinator does not move segments to decommissioning servers, effectively putting them in a type of "maintenance" mode. In this case, decommissioning servers do not participate in balancing or assignment by load rules. The Coordinator still considers segments on decommissioning servers as candidates to move or replicate on active servers.<br /><br />Decommissioning can stall if there are no available active servers to place the segments. You can use the maximum percent of decommissioning segment movements to prioritize balancing or to decrease commissioning time to prevent active servers from being overloaded. The value must be between 0 and 100.|70|
```
Proposed changes:
* Slight wording updates
* Makes BR tags self-closing as in https://github.com/apache/druid/pull/13107/files
